AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt

```diff
@@ -66,6 +70,44 @@ class ReviewerTest : RobolectricTest() {
         }
     }
 
+    @Test
+    fun testOnSelectedTags() {
+        // Add a note using basic model
+        addNoteUsingBasicModel()
+
+        // Start the Reviewer activity
+        val viewer = startRegularActivity<Reviewer>()
+
+        // Create a list of tags
+        val tags = listOf("tag1", "tag2")
+
+        // Define an arbitrary filter
+        val ARBITRARY_FILTER = CardStateFilter.DUE
+
+        // Assert that currentCard is not null before calling onSelectedTags
+        assertNotNull("currentCard should not be null", viewer.currentCard)
```

Comment:
My suggestion had two parts:
1- make an assertion against all possible state *prior* to changing state
2- make the assertion in an assertTrue style (if possible)

Per your and David's comments the method I just typed into the text box does not exist, but that's okay. Item 2 was a minor point

Item 1 is more important to me though. I think if you are going to assert on some state later (that viewer.isDisplayingAnswer has some value you expect), you had better assert that it has a known state prior to running the code under test. 

That's what I meant by asserting on all the the things you could.

Concretely then, with an API that apparently exists in a style that works (for legacy reasons) I mean add this assertion right here

```suggestion
        assertNotNull("currentCard should not be null", viewer.currentCard)
        assertTrue(!viewer.isDisplayingAnswer)
```

Reply:
Yeah about that, sorry mate I was traveling and couldn't add it in properly, was doing it on my phone remotely, and didn't read it properly 😅